test/support/src/helpers.h

```diff
@@ -759,6 +759,47 @@ std::string get_fragment_dir(std::string array_dir);
  */
 std::string get_commit_dir(std::string array_dir);
 
+/**
+ * Base class that lists the given path, ensures each directory has the
+ * expected file extensions, and stores the counts of each file type.
+ */
+class FileCount {
```

Comment:
Look at class SupportedFsS3 in the same folder and try to match the commenting style for your class. Same for CommitsDirectory.